### DISCUSSION TITLE
[Writing Rust code in helix]: Is there a way to show the type of the variable next to it.

### DISCUSSION THREAD

**Discussion by blitzerr** [helix-editor/helix#5201] [General] [answered]
In VS-Code, one can see the type of a variable or expression next to it (as shown below). Is there a way to enable such a thing in helix ?

I am assuming that the type is available from `rust-analyzer` but if someone can tell me how I can make helix render it inline, I would appreciate it a lot. 
<img width="1185" alt="image" src="https://user-images.githubusercontent.com/28721905/208281600-bc508b50-b845-469a-8fe3-3b7236f11cbd.png">

**Comment by gabydd** [comment, marked as answer]
Not yet, see #2070 and #411

**Reply by gabydd** [reply]
You can use ``<space>k`` which is ``hover`` to get types and documentation from the lsp

**Reply by blitzerr** [reply]
Thanks @gabydd for the space + k suggestion. That will have to do for now. 
 I will follow those issues very closely. 😀

**Comment by AddictArts** [comment]
Space K does nothing for me, on Windows, discovered when using for rustlings exercises. In a typical `cargo new` project now <space>+k is working. Wonder if I can build this latest on Windows, hmm.

**Reply by getreu** [reply]
[Feat: LSP Type Hints (#5934) · helix-editor/helix@bdcd4d9](https://github.com/helix-editor/helix/commit/bdcd4d9411655ab69245d803e88f88cc278127da) is available in the master branch since Feb 12.
